src/graphql/thegraph/apollo.ts

```diff
@@ -0,0 +1,40 @@
+import { ApolloClient, ApolloLink, concat, HttpLink, InMemoryCache } from '@apollo/client'
+import { SupportedChainId } from 'constants/chains'
+
+import store, { AppState } from '../../state/index'
+
+const CHAIN_SUBGRAPH_URL: Record<number, string> = {
+  [SupportedChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
+  [SupportedChainId.RINKEBY]: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
+
+  [SupportedChainId.ARBITRUM_ONE]: 'https://api.thegraph.com/subgraphs/name/ianlapham/arbitrum-minimal',
+
+  [SupportedChainId.OPTIMISM]: 'https://api.thegraph.com/subgraphs/name/ianlapham/optimism-post-regenesis',
+
+  [SupportedChainId.POLYGON]: 'https://api.thegraph.com/subgraphs/name/ianlapham/uniswap-v3-polygon',
+
+  [SupportedChainId.CELO]: 'https://api.thegraph.com/subgraphs/name/jesse-sawa/uniswap-celo',
+}
+
+const httpLink = new HttpLink({ uri: CHAIN_SUBGRAPH_URL[SupportedChainId.MAINNET] })
+
+// This middleware will allow us to dynamically update the uri for the requests based off chainId
+// For more information: https://www.apollographql.com/docs/react/networking/advanced-http-networking/
+const authMiddleware = new ApolloLink((operation, forward) => {
+  // add the authorization to the headers
+  const chainId = (store.getState() as AppState).application.chainId
+
+  operation.setContext(() => ({
+    uri:
+      chainId && CHAIN_SUBGRAPH_URL[chainId]
+        ? CHAIN_SUBGRAPH_URL[chainId]
+        : CHAIN_SUBGRAPH_URL[SupportedChainId.MAINNET],
```

Comment:
would this behave incorrectly if subgraph isn't defined for chain? should we throw?

alternatively you may be able to narrow the type of `CHAIN_SUBGRAPH_URL` with ts, `as const`

```
const CHAIN_SUBGRAPH_URL = {
  [SupportedChainId.MAINNET]: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
  [SupportedChainId.RINKEBY]: 'https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v3',
...
} as const
```

i *think* `CHAIN_SUBGRAPH_URL[chainId]` will never be undefined then

Reply:
This logic always falls back to Mainnet's chain. I also don't want to invest into changes to this logic if the long term plan is to rely on are in-house graphql data endpoint